File: ProjectEuler/010/010.cpp
#include <iostream>
#include <cmath>
using namespace std;

long prime_sieve(int n){
	bool primes[n]; primes[1] = false;
	for(int i = 2; i < n; i++) primes[i] = true;

	for(int i = 2; i < ceil(sqrt(n)); i++){
		if(primes[i]){
			for(long j = i*i; j < n; j+=i){
				primes[j] = false;
			}
		}
	}

	long sum = 0;
	for(int i = 0; i < n; i++){
		if(primes[i]) sum += i;
	}
	return sum;
}

int main(){
	int max = 2000000;

	cout << prime_sieve(max);
	
	return 0;
}

File: ProjectEuler/010/010.py
import math as m

def prime_sieve(n):
	primes = [True for i in range(n)]
	primes[1] = False

	for i in range(2, m.ceil(m.sqrt(n))):
		if(primes[i]):
			for j in range(i*i, n, i):
				primes[j] = False

	sum = 0
	for i in range(n):
		if(primes[i]):
			sum += i
	return sum

max = 2000000
print(prime_sieve(max))
